src/libraries/Common/src/Interop/Windows/SspiCli/SecuritySafeHandles.cs

```diff
@@ -431,34 +431,38 @@ internal abstract partial class SafeDeleteContext : SafeHandle
             SafeFreeContextBuffer outFreeContextBuffer = null;
             try
             {
-                Span<Interop.SspiCli.SecBuffer> inUnmanagedBuffer = stackalloc Interop.SspiCli.SecBuffer[inSecurityBufferDescriptor.cBuffers];
-                inUnmanagedBuffer.Clear();
+                // Allocate always maximum to allow better code optimization.
+                Span<Interop.SspiCli.SecBuffer> inUnmanagedBuffer = stackalloc Interop.SspiCli.SecBuffer[3];
+                for (int index = 0; index < inSecBuffers.Count; ++index)
```

Comment:
Can we assert anything about the relationship between inSecBuffers.Count and 3?  We appear to assume it's at least 3 below.